iothub/device/src/Pipeline/ErrorDelegatingHandler.cs

```diff
@@ -150,8 +150,7 @@ private async Task<T> ExecuteWithErrorHandlingAsync<T>(Func<Task<T>> asyncOperat
 
                     if (IsSecurityExceptionChain(ex))
                     {
-                        Exception innerException = (ex is IotHubClientException) ? ex.InnerException : ex;
```

Comment:
We'll want to hold on to this unwrapping logic. IsSecurityExceptionChain unwinds the entire exception chain to see if any exception underneath is a security exception. 
In case the top level exception is an IotHubClientException to begin with (because we did some transformation in the transport layer), then we'll have an IotHubClientException wrapping another IotHubClientException.